scripts/table_updates/update_table_schema.py

```diff
@@ -1,203 +1,412 @@
+                # extract columns that are not to be updated
+                tbl_schema.columnIds = [
+                    ele
+                    for ele in tbl_schema.columnIds
+                    if ele not in cols_to_update[table_id]["old"]
+                ]
+                # save updated columns to Synapse and get their ids
+                cols_to_update_new = [
+                    syn.store(i) for i in cols_to_update[table_id]["new"]
+                ]
+                cols_to_update_new_id = [col["id"] for col in cols_to_update_new]
+                tbl_schema.columnIds = tbl_schema.columnIds + cols_to_update_new_id
                 tbl_schema = syn.store(tbl_schema)
 
-def update_table_schema(syn, logger, dry_run):
+
+def update_table_schema(
+    syn: synapseclient.Synapse,
+    logger: logging,
+    dry_run: bool,
+    TABLE_INFO: dict[str, tuple],
+) -> None:
     # get the data elements
-    curated_data_element = download_synapse_table(syn,"syn21431364","dataType='curated'")
-    curated_data_element = curated_data_element[['variable','instrument','type','synColType','synColSize','numCols','colLabels']]
-    # create the master table 
-    sage_table_view = download_synapse_table(syn,TABLE_INFO['sage'][0],TABLE_INFO['sage'][1])
-    sage_table_view.drop(columns='table_type',axis=1,inplace=True)
-    bpc_table_view = download_synapse_table(syn,TABLE_INFO['bpc'][0],TABLE_INFO['bpc'][1])
-    bpc_table_view = bpc_table_view[['id','name']]
-    irr_table_view = download_synapse_table(syn,TABLE_INFO['irr'][0],TABLE_INFO['irr'][1])
-    irr_table_view = irr_table_view[['id','name']]
-    irr_table_view['name'] = irr_table_view['name'].apply(lambda x: x.replace(' - double curated',''))
-    master_table_view = pandas.merge(sage_table_view, 
-                                     pandas.merge(bpc_table_view,irr_table_view,
-                                                  on='name',suffixes=['_bpc','_irr']),
-                                     on='name')
-    # update table schema for Sage Internal tables
-    form_groups = master_table_view.groupby('form')
+    curated_data_element = download_synapse_table(
+        syn, table_id=TABLE_INFO["catalog_id"], condition="dataType='curated'"
+    )
+    curated_data_element = curated_data_element[
+        [
+            "variable",
+            "instrument",
+            "type",
+            "synColType",
+            "synColSize",
+            "numCols",
+            "colLabels",
+        ]
+    ]
+    # create the master table
+    sage_table_view = download_synapse_table(
+        syn, table_id=TABLE_INFO["sage"][0], condition=TABLE_INFO["sage"][1]
+    )
+    sage_table_view.drop(columns="table_type", axis=1, inplace=True)
+    bpc_table_view = download_synapse_table(
+        syn, table_id=TABLE_INFO["bpc"][0], condition=TABLE_INFO["bpc"][1]
+    )
+    bpc_table_view = bpc_table_view[["id", "name"]]
+    ## comment out irr for now in case it is needed for phase 3
```

Comment:
Commented out for now because we are not sure if we will need to update irr tables in phase 3.